### PR TITLE
backport: use GitHub actions (linux support) (#236)

### PR DESCRIPTION
**Pull request by v1v**
Backport #236

### CODE REVIEW

**Comment by Mpdreamz** [comment]
https://github.com/elastic/ecs-dotnet/actions/runs/4064290493/jobs/6997544409 this seems odd it should not publish prereleases like this to nuget Unless its building from a tag.

**Comment by Mpdreamz** [comment]
https://github.com/elastic/ecs-dotnet/releases/tag/1.6.0-alpha1.50 should have not been created.

**Comment by v1v** [comment]
> https://github.com/elastic/ecs-dotnet/actions/runs/4064290493/jobs/6997544409 this seems odd it should not publish prereleases like this to nuget Unless its building from a tag.

https://github.com/elastic/ecs-dotnet/actions/runs/4064290493/workflow#L5 even the build should have not been triggered 

The reason seems to be 

image


https://github.com/elastic/ecs-dotnet/commit/c9d84f297331595af9b40b29b0192e2ba182dbd6

I wonder whether the `on: create` is not correct, let me double check

**Comment by v1v** [comment]
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#create 

> Runs your workflow when someone creates a Git reference (Git branch or tag) in the workflow's repository.

And it seems the filter is not honoured as per https://github.com/actions/runner/issues/1007#issuecomment-808904408

**Comment by v1v** [comment]
> https://github.com/elastic/ecs-dotnet/releases/tag/1.6.0-alpha1.50 should have not been created.


What do we do?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+env:
+  # This is the version to be used for both OS platforms (windows and linux)
+  DOTNET_VERSION: '6.0.400'
```
Comment (v1v) [inline .github/workflows/test.yml:27]:
```suggestion
  DOTNET_VERSION: '5.0.100'
```

**Comment by Mpdreamz** [comment]
I deleted the tag/release. Since the actual publish to nuget failed there is not other cleanup needed. 

We might need to generate a new key to be able to publish again.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+       fetch-depth: 0
+
+    - name: Run release -c false within the docker container
+      run: |
+        docker run --rm -t \
+            -u "$(id -u):$(id -g)" \
+            -v $(pwd):/src \
+            -w /src \
+            -e HOME=/tmp \
+            mcr.microsoft.com/dotnet/sdk:$DOTNET_VERSION-jammy \
```
Comment (v1v) [inline .github/workflows/test.yml:76]:
```suggestion
            mcr.microsoft.com/dotnet/sdk:$DOTNET_VERSION \
```

**Comment by v1v** [comment]
> We might need to generate a new key to be able to publish again.

I might need to find how to do it, do you happen to remember how?

**Pull request closed without merging**